extensions/flutter-pub-dev-search/src/index.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  ActionPanel,
+  CopyToClipboardAction,
+  List,
+  OpenInBrowserAction,
+  showToast,
+  ToastStyle,
+  randomId,
+} from "@raycast/api";
+import { useState, useEffect, useRef } from "react";
+import fetch, { AbortError } from "node-fetch";
+
+export default function Command() {
+  const { state, search } = useSearch();
+
+  return (
+    <List isLoading={state.isLoading} onSearchTextChange={search} searchBarPlaceholder="Search for a Dart/Flutter package..." throttle>
+      <List.Section title="Results" subtitle={state.results.length + ""}>
+        {state.results.map((searchResult) => (
+          <SearchListItem key={searchResult.id} searchResult={searchResult} />
+        ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({ searchResult }: { searchResult: SearchResult }) {
+  return (
+    <List.Item
+      title={searchResult.name}
+      // subtitle={searchResult.description}
+      // accessoryTitle={searchResult.version}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <CopyToClipboardAction
+              title="Copy Install Command"
+              content={`flutter pub add ${searchResult.name}`}
+              onCopy={(content) => showToast( ToastStyle.Success,`${content} Copied to clipboard!\n Now go and paste the command into your IDE's terminal to add the package.`)}
+            />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
+            <OpenInBrowserAction 
+            title="Open in Browser" 
+            url={searchResult.url} 
+            shortcut={{ modifiers: ["cmd"], key: "." }}
+
+            />
+          </ActionPanel.Section>
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+function useSearch() {
+  const [state, setState] = useState<SearchState>({ results: [], isLoading: true });
+  const cancelRef = useRef<AbortController | null>(null);
+
+  useEffect(() => {
+    search("");
+    return () => {
+      cancelRef.current?.abort();
+    };
+  }, []);
+
+  async function search(searchText: string) {
+    cancelRef.current?.abort();
+    cancelRef.current = new AbortController();
+    try {
+      setState((oldState) => ({
+        ...oldState,
+        isLoading: true,
+      }));
+      const results = await performSearch(searchText, cancelRef.current.signal);
+      setState((oldState) => ({
+        ...oldState,
+        results: results,
+        isLoading: false,
+      }));
+    } catch (error) {
+      if (error instanceof AbortError) {
+        return;
+      }
+      console.error("search error", error);
+      showToast(ToastStyle.Failure, "Could not perform search", String(error));
+    }
+  }
+
+  return {
+    state: state,
+    search: search,
+  };
+}
+
+async function performSearch(searchText: string, signal: AbortSignal): Promise<SearchResult[]> {
+  console.log(searchText);
+  const params = new URLSearchParams();
+  params.append("q", searchText.length === 0 ? "http" : searchText);
```

Comment:
Instead of defaulting to "http" as first search term, is there a way to get the list of favorites or most popular packages through the API? It might make more sense to show that by default rather searching for "http".

Reply:
`https://pub.dev/api/search?q=` will give the list of popular/top packages. So, I have removed the "http".